### PR TITLE
[FIX] `first_level_from_bids` will now return subjects in order

### LINKED ISSUE #4581: `first_level_from_bids` should return subjects in order
If subjects labels are passe, then `first_level_from_bids` will loop over subjects in a random order (uses `set` to avoid duplicates):

https://github.com/nilearn/nilearn/blob/51778dd58d78036afdef7b47ae3137ff0dbda290/nilearn/glm/first_level/first_level.py#L1684

But if no subjects are passed then they are run in a sorted manner:

https://github.com/nilearn/nilearn/blob/51778dd58d78036afdef7b47ae3137ff0dbda290/nilearn/glm/first_level/first_level.py#L1669

It'd be preferable if the behavior is consistent.

Brought to my attention by @TrevorKMDay a while ago but forgot to fix it.

> [!TIP]
> # **💻 🍁  For the [hacktoberfest](https://hacktoberfest.com/) contributors 🍁 💻**
>
> Make sure to:
> - read our [contributing guidelines](https://nilearn.github.io/stable/development.html#contribution-guidelines)
>
> If this is your first time contributing to Nilearn, add your details in the [CITATION.cff file](https://github.com/nilearn/nilearn/blob/dcd92aa357ab3a751161aeea2edfa3a775ae644f/CITATION.cff#L42).
>
> For your pull request to be valid, you should do the following:

- [ ] Remove this line (only sort things once at the return statement):

https://github.com/nilearn/nilearn/blob/51778dd58d78036afdef7b47ae3137ff0dbda290/nilearn/glm/first_level/first_level.py#L1669C22-L1669C51

- [ ] Replace the following line

https://github.com/nilearn/nilearn/blob/51778dd58d78036afdef7b47ae3137ff0dbda290/nilearn/glm/first_level/first_level.py#L1684

By:

```python
return sorted(set(sub_labels))
```

- [ ] add a test to make sure subject are always returned sorted

### PR DESCRIPTION
**Pull request by tharun634**
- Closes #4581

### CODE REVIEW

**Review by Remi-Gau — CHANGES_REQUESTED** [review]
Had initially forgotten to mention it in the original issue but do you think you can add a test to make sure that subjects are always returned in a sorted order.

See the test here:
https://github.com/nilearn/nilearn/blob/dcd92aa357ab3a751161aeea2edfa3a775ae644f/nilearn/glm/tests/test_first_level.py#L1483

**Comment by Remi-Gau** [comment]
Don't worry about the last failure from ruff we have got it also on the main branch so it is not related to your PR.
Another PR is ready to fix it.

**Review by Remi-Gau — APPROVED** [review]
Ok this looks good to me.
Thanks for taking that one @tharun634

Juat need another review from @bthirion or @man-shu.

File: nilearn/glm/tests/test_first_level.py
Comment (bthirion) [inline nilearn/glm/tests/test_first_level.py:2159]: The error should not be triggered, right ?

File: nilearn/glm/tests/test_first_level.py
Comment (bthirion) [inline nilearn/glm/tests/test_first_level.py:2186]: Same here.

**Review by bthirion — COMMENTED** [review]
LGTM otherwise

File: nilearn/glm/tests/test_first_level.py
Comment (tharun634) [reply nilearn/glm/tests/test_first_level.py:2159]: No, this is just an error message in case of test failure.

Let me know if you want me to remove the failure message.

Comment (bthirion) [reply nilearn/glm/tests/test_first_level.py:2159]: I prefer to remove it, because it's not supposed to be triggered.

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by Remi-Gau** [comment]
Waiting for the CI runs to finish but this should be good to go.

**Pull request merged**
